docs/src/explanations/chisel-enum.md

```diff
@@ -167,6 +168,25 @@ val (log2, _) = grabLog(ChiselStage.emitChirrtl(new SafeFromUInt))
 println(s"```\n$log2```")
 ```
 
+You can also suppress the warning by using `suppressEnumCastWarning`. This is
+primarily used for casting from [[UInt]] to a Bundle type that contains an
+Enum, where the [[UInt]] is known to be valid for the Bundle type.
+
+```scala mdoc
+class MyBundle extends Bundle {
+  val addr = UInt(8.W)
+  val op = Opcode()
+}
+
+class SuppressedFromUInt extends Module {
+  val in = IO(Input(UInt(15.W)))
+  val out = IO(Output(new MyBundle()))
+  suppressEnumCastWarning {
+    out := in.asTypeOf(new MyBundle)
+  }
+}
+```
+
```

Comment:
```scala mdoc:invisible

val (log3, _) = grabLog(ChiselStage.emitChirrtl(new SuppressedFromUInt))
assert(log3.isEmpty)
```

Sorry, one last suggestion but it would be nice to actually check that there's no when generating this, blending testing with documentation but still nice to check the example.